dev/release/src/chart.ts

```diff
@@ -0,0 +1,30 @@
+import { readFileSync } from 'fs'
+
+import { load as loadYAML } from 'js-yaml'
+import * as semver from 'semver'
+
+export interface Metadata {
+    apiVersion: string
+    name: string
+    description: string
+    type: string
+    version: semver.SemVer
+    appVersion: semver.SemVer
+}
+
+export function parseChartMetadata(chartYamlPath: string): Metadata {
+    const chartYamlContents = readFileSync(chartYamlPath, 'utf8').toString()
+    const chartYaml = loadYAML(chartYamlContents) as any // eslint-disable-line @typescript-eslint/no-unsafe-assignment,@typescript-eslint/no-explicit-any
```

Comment:
I would just leave `version` and `appVersion` as `string` and do:

```suggestion
    return loadYAML(chartYamlContents) as Metadata
```

Then, handle the parsing later on. This is what we do for the release config: https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/dev/release/src/config.ts?L17-18

This makes it more explicit what the written config will look like at the end of the day, instead of relying on `semver.Semver` automagic stringifying